vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridDataProviderConstructorTest.java

```diff
@@ -0,0 +1,354 @@
+        Grid<Person> grid = new Grid<>(dataProvider);
+        return createColumns(grid);
+    }
+
+    private final Grid<Person> createGrid(Collection<Person> items) {
+        Grid<Person> grid = new Grid<>(items);
+        return createColumns(grid);
+    }
+
+    private final Grid<Person> createColumns(Grid<Person> grid) {
+        grid.addColumn(Person::getFirstName).setHeader("First name")
+                .setKey("first");
+        grid.addColumn(Person::getLastName).setHeader("Last name")
+                .setKey("last");
+        grid.addColumn(Person::getEmail).setHeader("Email").setKey("email");
+        grid.addColumn(Person::getProfession).setHeader("Profession")
+                .setKey("profession");
+        grid.addColumn(new LocalDateRenderer<>(Person::getBirthday,
+                        () -> DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM)))
+                .setHeader("Date Renderer").setKey("date");
+        grid.addColumn(
+                        new LocalDateRenderer<>(Person::getBirthday, "dd/MM/yyyy"))
+                .setHeader("String Renderer").setKey("renderer");
+        return grid;
+    }
+
+    public final static class Person {
+
+        private String firstName;
+
+        private String lastName;
+
+        private String email;
+
+        private LocalDate birthday;
+
+        private Integer id;
+
+        private String profession;
+
+        public Person(Integer id, String firstName, String lastName,
+                      String email, LocalDate birthday, String profession) {
+            this.firstName = firstName;
+            this.lastName = lastName;
+            this.email = email;
+            this.birthday = birthday;
+            this.id = id;
+            this.profession = profession;
+        }
+
+        public String getFirstName() {
+            return firstName;
+        }
+
+        public void setFirstName(String firstName) {
+            this.firstName = firstName;
+        }
+
+        public String getLastName() {
+            return lastName;
+        }
+
+        public void setLastName(String lastName) {
+            this.lastName = lastName;
+        }
+
+        public String getEmail() {
+            return email;
+        }
+
+        public void setEmail(String email) {
+            this.email = email;
+        }
+
+        public LocalDate getBirthday() {
+            return birthday;
+        }
+
+        public void setBirthday(LocalDate birthday) {
+            this.birthday = birthday;
+        }
+
+        public Integer getId() {
+            return id;
+        }
+
+        public void setId(Integer id) {
+            this.id = id;
+        }
+
+        public String getProfession() {
+            return profession;
+        }
+
+        public void setProfession(String profession) {
+            this.profession = profession;
+        }
+    }
+
+    private final class PersonBackendDataProvider
```

Comment:
All the custom data providers can extend from `AbstractDataProvider`, which allows removing some of the overrides:
- `refreshItem`
- `refreshItems`
- `refreshAll`
- `addDataProviderListener`